components/nimbus/src/metrics.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    enrollment::ExperimentEnrollment, glean_metrics::nimbus_events::EnrollmentStatusExtra,
+    EnrollmentStatus,
+};
+use serde_derive::{Deserialize, Serialize};
+
+pub trait HostMetricsHandler: Send + Sync {
+    fn record_enrollment_status(&self, enrollment_status_extra: EnrollmentStatusExtraDef);
+
+    /// Only used for testing
+    fn get_enrollment_status_records(&self) -> Vec<EnrollmentStatusExtraDef>;
+
+    /// Only used for testing
+    fn clear(&self);
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(remote = "EnrollmentStatusExtra")]
+pub struct EnrollmentStatusExtraDef {
+    pub branch: Option<String>,
+    pub conflict_slug: Option<String>,
+    pub error_string: Option<String>,
+    pub reason: Option<String>,
+    pub slug: Option<String>,
+    pub status: Option<String>,
+}
+
+impl From<ExperimentEnrollment> for EnrollmentStatusExtraDef {
+    fn from(enrollment: ExperimentEnrollment) -> Self {
+        let mut branch_value: Option<String> = None;
+        let mut reason_value: Option<String> = None;
+        let mut error_value: Option<String> = None;
+        match &enrollment.status {
+            EnrollmentStatus::Enrolled { reason, branch, .. } => {
```

Comment:
So idk if there's a way I could cast `reason` to a string from its enum type right here but it'd be nice because I could fold two of these branches up into one.